diff --git a/arch/arm64/boot/dts/rockchip/overlay/rk3588-can0-canfd.dts b/arch/arm64/boot/dts/rockchip/overlay/rk3588-can0-canfd.dts
new file mode 100644
index 000000000000..ec411d1bb9bd
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/overlay/rk3588-can0-canfd.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "rockchip,rk3588";
+
+    fragment@0 {
+        target = <&can0>;
+        __overlay__ {
+            compatible = "rockchip,canfd-1.0";
+        };
+    };
+};
diff --git a/arch/arm64/boot/dts/rockchip/overlay/rk3588-can1-canfd.dts b/arch/arm64/boot/dts/rockchip/overlay/rk3588-can1-canfd.dts
new file mode 100644
index 000000000000..759e278008dc
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/overlay/rk3588-can1-canfd.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "rockchip,rk3588";
+
+    fragment@0 {
+        target = <&can1>;
+        __overlay__ {
+            compatible = "rockchip,canfd-1.0";
+        };
+    };
+};
diff --git a/arch/arm64/boot/dts/rockchip/overlay/rk3588-can2-can20.dts b/arch/arm64/boot/dts/rockchip/overlay/rk3588-can2-can20.dts
new file mode 100644
index 000000000000..7923f233e96b
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/overlay/rk3588-can2-can20.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "rockchip,rk3588";
+
+    fragment@0 {
+        target = <&can2>;
+        __overlay__ {
+            compatible = "rockchip,can-2.0";
+        };
+    };
+};
