diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-cm5-tablet.dts b/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-cm5-tablet.dts
index da52879906c0..186902bd0d72 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-cm5-tablet.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-cm5-tablet.dts
@@ -171,10 +171,8 @@ pcie20_avdd1v8: pcie20-avdd1v8 {
 	vcc5v0_host: vcc5v0-host {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc5v0_host";
-		regulator-boot-on;
-		regulator-always-on;
 		enable-active-high;
-		gpio = <&gpio4 RK_PA2 GPIO_ACTIVE_HIGH>;
+		gpio = <&gpio4 RK_PA4 GPIO_ACTIVE_HIGH>;
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 		vin-supply = <&vcc5v0_usb>;
@@ -187,11 +185,12 @@ vbus5v0_typec: vbus5v0-typec {
 		regulator-name = "vbus5v0_typec";
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
-		enable-active-high;
-		gpio = <&gpio4 RK_PA4 GPIO_ACTIVE_HIGH>;
-		vin-supply = <&vcc5v0_usb>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&typec5v_pwren>;
+		// enable-active-high;
+		regulator-always-on;
+    	regulator-boot-on;
+    	// -supply = <&vcc5v0_usb>;
+		// pinctrl-names = "default";
+		// pinctrl-0 = <&typec5v_pwren>;
 	};
 
 	sdio_pwrseq: sdio-pwrseq {
@@ -232,70 +231,8 @@ wireless_wlan: wireless-wlan {
 		WIFI,host_wake_irq = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
 		WIFI,poweren_gpio = <&gpio0 RK_PD0 GPIO_ACTIVE_HIGH>;
 		status = "okay";
-	};
-
-	es8388_sound: es8388-sound {
-		status = "okay";
-		compatible = "rockchip,multicodecs-card";
-		rockchip,card-name = "rockchip,es8388";
-		hp-det-gpio = <&gpio3 RK_PC6 GPIO_ACTIVE_HIGH>;
-		io-channels = <&saradc 3>;
-		io-channel-names = "adc-detect";
-		keyup-threshold-microvolt = <1800000>;
-		poll-interval = <100>;
-		hp-con-gpio = <&gpio0 RK_PC2 GPIO_ACTIVE_HIGH>;
-		rockchip,aux-devs = <&aw87xxx_pa1 &aw87xxx_pa2>;
-		rockchip,format = "i2s";
-		rockchip,mclk-fs = <256>;
-		rockchip,cpu = <&i2s2_2ch>;
-		rockchip,codec = <&es8388>;
-		rockchip,audio-routing =
-			"Headphone", "LOUT1",
-			"Headphone", "ROUT1",
-			"Headphone", "Headphone Power",
-			"Headphone", "Headphone Power",
-			"LINPUT1", "Main Mic",
-			"LINPUT2", "Main Mic",
-			"RINPUT1", "Headset Mic",
-			"RINPUT2", "Headset Mic";
-		pinctrl-names = "default";
-		pinctrl-0 = <&hp_det>;
-		play-pause-key {
-			label = "playpause";
-			linux,code = <KEY_PLAYPAUSE>;
-			press-threshold-microvolt = <2000>;
-		};
-	};
-
-	leds: gpio-leds {
-		compatible = "gpio-leds";
-		pinctrl-names = "default";
-		pinctrl-0 =<&led_gpio>;
-		status = "okay";
-
-		led@1 {
-			gpios = <&gpio4 RK_PB6 GPIO_ACTIVE_HIGH>;
-			label = "status_led";
-			linux,default-trigger = "heartbeat";
-			linux,default-trigger-delay-ms = <0>;
-		};
-	};
-
-	fan: pwm-fan {
-		compatible = "pwm-fan";
-		#cooling-cells = <2>;
-		pwms = <&pwm15 0 50000 0>;
-		cooling-levels = <0 50 100 150 200 255>;
-		rockchip,temp-trips = <
-			40000   2
-			50000   3
-			55000   4
-			60000   5
-			65000   5
-			70000   5
-		>;
-
-		status = "okay";
+		// clocks = <&hym8563>;
+		// clock-names = "ref_wifi_clk";
 	};
 
 	ext_cam_ov5647_clk: external-camera-ov5647-clock {
@@ -304,6 +241,46 @@ ext_cam_ov5647_clk: external-camera-ov5647-clock {
 		clock-output-names = "ext_cam_ov5647_clk";
 		#clock-cells = <0>;
 	};
+
+	fan0: gpio-fan-0 {
+        compatible = "gpio-fan";
+        gpios = <&gpio1 RK_PB7 GPIO_ACTIVE_HIGH>;
+        gpio-fan,speed-map = <0 0 5000 1>; /* 0: 关闭, 5000: 开启 */
+        #cooling-cells = <2>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&fan0_en>;
+        status = "okay";
+    };
+
+    fan1: gpio-fan-1 {
+        compatible = "gpio-fan";
+        gpios = <&gpio1 RK_PB6 GPIO_ACTIVE_HIGH>;
+        gpio-fan,speed-map = <0 0 5000 1>;
+        #cooling-cells = <2>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&fan1_en>;
+        status = "okay";
+    };
+
+    fan2: gpio-fan-2 {
+        compatible = "gpio-fan";
+        gpios = <&gpio1 RK_PA3 GPIO_ACTIVE_HIGH>;
+        gpio-fan,speed-map = <0 0 5000 1>;
+        #cooling-cells = <2>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&fan2_en>;
+        status = "okay";
+    };
+
+    fan3: gpio-fan-3 {
+        compatible = "gpio-fan";
+        gpios = <&gpio1 RK_PD2 GPIO_ACTIVE_HIGH>;
+        gpio-fan,speed-map = <0 0 5000 1>;
+        #cooling-cells = <2>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&fan3_en>;
+        status = "okay";
+    };
 };
 
 &av1d_mmu {
@@ -452,7 +429,7 @@ &saradc {
 };
 
 &sdio {
-	max-frequency = <150000000>;
+	max-frequency = <50000000>;
 	no-sd;
 	no-mmc;
 	bus-width = <4>;
@@ -581,303 +558,24 @@ &spdif_tx2{
 	status = "okay";
 };
 
-&i2c0 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c0m2_xfer>;
-
-	vdd_cpu_big0_s0: vdd_cpu_big0_mem_s0: rk8602@42 {
-		compatible = "rockchip,rk8602";
-		reg = <0x42>;
-		vin-supply = <&vcc5v0_sys>;
-		regulator-compatible = "rk860x-reg";
-		regulator-name = "vdd_cpu_big0_s0";
-		regulator-min-microvolt = <550000>;
-		regulator-max-microvolt = <1050000>;
-		regulator-ramp-delay = <2300>;
-		rockchip,suspend-voltage-selector = <1>;
-		regulator-boot-on;
-		regulator-always-on;
-		regulator-state-mem {
-			regulator-off-in-suspend;
-		};
-	};
-
-	vdd_cpu_big1_s0: vdd_cpu_big1_mem_s0: rk8603@43 {
-		compatible = "rockchip,rk8603";
-		reg = <0x43>;
-		vin-supply = <&vcc5v0_sys>;
-		regulator-compatible = "rk860x-reg";
-		regulator-name = "vdd_cpu_big1_s0";
-		regulator-min-microvolt = <550000>;
-		regulator-max-microvolt = <1050000>;
-		regulator-ramp-delay = <2300>;
-		rockchip,suspend-voltage-selector = <1>;
-		regulator-boot-on;
-		regulator-always-on;
-		regulator-state-mem {
-			regulator-off-in-suspend;
-		};
-	};
-};
-
-&i2c2 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c2m0_xfer>;
-
-	vdd_npu_s0: vdd_npu_mem_s0: rk8602@42 {
-		compatible = "rockchip,rk8602";
-		reg = <0x42>;
-		vin-supply = <&vcc5v0_sys>;
-		regulator-compatible = "rk860x-reg";
-		regulator-name = "vdd_npu_s0";
-		regulator-min-microvolt = <550000>;
-		regulator-max-microvolt = <950000>;
-		regulator-ramp-delay = <2300>;
-		rockchip,suspend-voltage-selector = <1>;
-		regulator-boot-on;
-		regulator-always-on;
-		regulator-state-mem {
-			regulator-off-in-suspend;
-		};
-	};
-};
-
-&i2c3 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c3m1_xfer>;
-
-	es8388: es8388@11 {
-		status = "okay";
-		#sound-dai-cells = <0>;
-		compatible = "everest,es8388", "everest,es8323";
-		reg = <0x11>;
-		clocks = <&mclkout_i2s2>;
-		clock-names = "mclk";
-		assigned-clocks = <&mclkout_i2s2>;
-		assigned-clock-rates = <12288000>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&i2s2m1_mclk>;
-	};
-};
-
-&i2c6 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c6m3_xfer>;
-
-	sc89890: sc89890@6a {
-		compatible = "sc,sc89890";
-		reg = <0x6a>;
-		status = "okay";
-		pinctrl-names = "default";
-		pinctrl-0 = <&charger_ok>;
-
-		interrupt-parent = <&gpio1>;
-		interrupts = <RK_PC0 IRQ_TYPE_EDGE_FALLING>;
-		//otg-mode-en-gpios = <&gpio1 RK_PC1 GPIO_ACTIVE_HIGH>;
-
-		sc,battery-regulation-voltage = <4350000>; /* 4.4V */
-		sc,charge-current = <4000000>; /* 4.0A */
-		sc,termination-current = <100000>;  /* 100mA */
-		sc,precharge-current = <500000>; /* 130mA */
-		sc,minimum-sys-voltage = <3400000>; /* 3V */
-		sc,thermal-regulation-threshold = <4000000>;
-		sc,ibatcomp-clamp-microvolt = <100000>;
-		sc,ibatcomp-micro-ohms = <20>;
-	};
-
-	usbc0: fusb302@22 {
-		compatible = "fcs,fusb302";
-		reg = <0x22>;
-		interrupt-parent = <&gpio4>;
-		interrupts = <RK_PA3 IRQ_TYPE_LEVEL_LOW>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&usbc0_int>;
-		vbus-supply = <&vbus5v0_typec>;
-		status = "okay";
-
-		ports {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			port@0 {
-				reg = <0>;
-				usbc0_role_sw: endpoint@0 {
-					remote-endpoint = <&dwc3_0_role_switch>;
-				};
-			};
-		};
-
-		usb_con: connector {
-			compatible = "usb-c-connector";
-			label = "USB-C";
-			data-role = "dual";
-			power-role = "dual";
-			try-power-role = "sink";
-			op-sink-microwatt = <1000000>;
-			sink-pdos =
-				<PDO_FIXED(5000, 1000, PDO_FIXED_USB_COMM)>;
-			source-pdos =
-				<PDO_FIXED(5000, 3000, PDO_FIXED_USB_COMM)>;
-
-			altmodes {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				altmode@0 {
-					reg = <0>;
-					svid = <0xff01>;
-					vdo = <0xffffffff>;
-				};
-			};
-
-			ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				port@0 {
-					reg = <0>;
-					usbc0_orien_sw: endpoint {
-						remote-endpoint = <&usbdp_phy0_orientation_switch>;
-					};
-				};
-
-				port@1 {
-					reg = <1>;
-					dp_altmode_mux: endpoint {
-						remote-endpoint = <&usbdp_phy0_dp_altmode_mux>;
-					};
-				};
-			};
-		};
-	};
-};
-
-&i2c7 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c7m3_xfer>;
-
-	hym8563: hym8563@51 {
-		compatible = "haoyu,hym8563";
-		reg = <0x51>;
-		#clock-cells = <0>;
-		clock-frequency = <32768>;
-		clock-output-names = "hym8563";
-		pinctrl-names = "default";
-		pinctrl-0 = <&hym8563_int>;
-		interrupt-parent = <&gpio0>;
-		interrupts = <RK_PB0 IRQ_TYPE_LEVEL_LOW>;
-		status = "okay";
-	};
-};
-
-&i2s2_2ch {
-	status = "okay";
-	pinctrl-0 = <&i2s2m1_lrck
-	             &i2s2m1_sclk
-	             &i2s2m1_sdi
-	             &i2s2m1_sdo>;
-};
-
-&i2s5_8ch {
-	status = "okay";
-};
-
 &sata0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&sata_reset>;
 	status = "disabled";
 };
 
+//phy2
+&pcie2x1l1 {
+	reset-gpios = <&gpio4 RK_PA2 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
 //phy0
 &pcie2x1l2 {
 	reset-gpios = <&gpio3 RK_PD1 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 };
 
-&pinctrl {
-	usb-typec {
-		usbc0_int: usbc0-int {
-			rockchip,pins = <4 RK_PA3 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-
-		typec5v_pwren: typec5v-pwren {
-			rockchip,pins = <4 RK_PA4 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-
-	usb {
-		vcc5v0_host_en: vcc5v0-host-en {
-			rockchip,pins = <4 RK_PA2 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-
-	sdio-pwrseq {
-		wifi_enable_h: wifi-enable-h {
-			rockchip,pins = <0 RK_PD0 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-
-	headphone {
-		hp_det: hp-det {
-			rockchip,pins = <3 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-
-	hym8563 {
-		hym8563_int: hym8563-int {
-			rockchip,pins = <0 RK_PB0 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-	};
-
-	wireless-bluetooth {
-		uart9_gpios: uart9-gpios {
-			rockchip,pins = <3 RK_PD2 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-
-		bt_reset_gpio: bt-reset-gpio {
-			rockchip,pins = <0 RK_PD3 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-
-		bt_wake_gpio: bt-wake-gpio {
-			rockchip,pins = <0 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-
-		bt_irq_gpio: bt-irq-gpio {
-			rockchip,pins = <0 RK_PC5 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-	};
-
-	wireless-wlan {
-		wifi_host_wake_irq: wifi-host-wake-irq {
-			rockchip,pins = <0 RK_PA0 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-	};
-
-	leds {
-		led_gpio: led-gpio {
-			rockchip,pins = <4 RK_PB6 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-
-	sata {
-		sata_reset:sata-reset{
-			rockchip,pins = <3 RK_PD1 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-	};
-
-	charger {
-		charger_ok: charger_ok {
-			rockchip,pins = <1 RK_PC0 RK_FUNC_GPIO &pcfg_pull_up>,
-			                <1 RK_PC1 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-	};
-};
-
 &u2phy0 {
 	status = "okay";
 };
@@ -892,7 +590,8 @@ &u2phy3 {
 
 &u2phy0_otg {
 	status = "okay";
-	rockchip,typec-vbus-det;
+	// phy-supply = <&vcc5v0_host>;
+	// rockchip,typec-vbus-det;
 };
 
 &u2phy2_host {
@@ -925,21 +624,13 @@ &usbdrd3_0 {
 
 &usbdrd_dwc3_0 {
 	status = "okay";
-	dr_mode = "otg";
-	usb-role-switch;
-	port {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		dwc3_0_role_switch: endpoint@0 {
-			reg = <0>;
-			remote-endpoint = <&usbc0_role_sw>;
-		};
-	};
+	dr_mode = "peripheral";
+	maximum-speed = "high-speed";
+	extcon = <&u2phy0>;
 };
 
 &usbhost3_0 {
-	dr_mode = "host";
-	status = "okay";
+	status = "disabled";
 };
 
 &usbhost_dwc3_0 {
@@ -947,23 +638,7 @@ &usbhost_dwc3_0 {
 };
 
 &usbdp_phy0 {
-	orientation-switch;
-	svid = <0xff01>;
 	status = "okay";
-
-	port {
-		#address-cells = <1>;
-		#size-cells = <0>;
-		usbdp_phy0_orientation_switch: endpoint@0 {
-		reg = <0>;
-			remote-endpoint = <&usbc0_orien_sw>;
-		};
-
-		usbdp_phy0_dp_altmode_mux: endpoint@1 {
-			reg = <1>;
-			remote-endpoint = <&dp_altmode_mux>;
-		};
-	};
 };
 
 &usbdp_phy0_dp {
@@ -1041,41 +716,138 @@ &sfc {
 	status = "disabled";
 };
 
-&pwm0 {
-	pinctrl-names = "active";
-	pinctrl-0 = <&pwm0m1_pins>;
+&i2c0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c0m2_xfer>;
+
+	vdd_cpu_big0_s0: vdd_cpu_big0_mem_s0: rk8602@42 {
+		compatible = "rockchip,rk8602";
+		reg = <0x42>;
+		vin-supply = <&vcc5v0_sys>;
+		regulator-compatible = "rk860x-reg";
+		regulator-name = "vdd_cpu_big0_s0";
+		regulator-min-microvolt = <550000>;
+		regulator-max-microvolt = <1050000>;
+		regulator-ramp-delay = <2300>;
+		rockchip,suspend-voltage-selector = <1>;
+		regulator-boot-on;
+		regulator-always-on;
+		regulator-state-mem {
+			regulator-off-in-suspend;
+		};
+	};
+
+	vdd_cpu_big1_s0: vdd_cpu_big1_mem_s0: rk8603@43 {
+		compatible = "rockchip,rk8603";
+		reg = <0x43>;
+		vin-supply = <&vcc5v0_sys>;
+		regulator-compatible = "rk860x-reg";
+		regulator-name = "vdd_cpu_big1_s0";
+		regulator-min-microvolt = <550000>;
+		regulator-max-microvolt = <1050000>;
+		regulator-ramp-delay = <2300>;
+		rockchip,suspend-voltage-selector = <1>;
+		regulator-boot-on;
+		regulator-always-on;
+		regulator-state-mem {
+			regulator-off-in-suspend;
+		};
+	};
+};
+
+&i2c1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c1m2_xfer>;
+
+	hym8563: hym8563@51 {
+		compatible = "haoyu,hym8563";
+		reg = <0x51>;
+		#clock-cells = <0>;
+		clock-frequency = <32768>;
+		clock-output-names = "hym8563";
+		pinctrl-names = "default";
+		pinctrl-0 = <&hym8563_int>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <RK_PB0 IRQ_TYPE_LEVEL_LOW>;
+		status = "okay";
+	};
+};
+
+&i2c2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c2m0_xfer>;
+
+	vdd_npu_s0: vdd_npu_mem_s0: rk8602@42 {
+		compatible = "rockchip,rk8602";
+		reg = <0x42>;
+		vin-supply = <&vcc5v0_sys>;
+		regulator-compatible = "rk860x-reg";
+		regulator-name = "vdd_npu_s0";
+		regulator-min-microvolt = <550000>;
+		regulator-max-microvolt = <950000>;
+		regulator-ramp-delay = <2300>;
+		rockchip,suspend-voltage-selector = <1>;
+		regulator-boot-on;
+		regulator-always-on;
+		regulator-state-mem {
+			regulator-off-in-suspend;
+		};
+	};
+};
+
+&i2c3 {
 	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c3m1_xfer>;
 };
 
-&pwm1 {
-	pinctrl-names = "active";
-	pinctrl-0 = <&pwm1m2_pins>;
+&i2c4 {
 	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c4m3_xfer>;
 };
 
-&pwm13 {
-	pinctrl-names = "active";
-	pinctrl-0 = <&pwm13m2_pins>;
+&i2c5 {
 	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c5m3_xfer>;
 };
 
-&pwm14 {
-	pinctrl-names = "active";
-	pinctrl-0 = <&pwm14m0_pins>;
+&i2c6 {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c6m3_xfer>;
+};
+
+&i2c7 {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c7m3_xfer>;
+};
+
+&i2s2_2ch {
 	status = "okay";
+	pinctrl-0 = <&i2s2m1_lrck
+	             &i2s2m1_sclk
+	             &i2s2m1_sdi
+	             &i2s2m1_sdo>;
 };
 
-&pwm15 {
-	pinctrl-names = "active";
-	pinctrl-0 = <&pwm15m2_pins>;
+&i2s5_8ch {
 	status = "okay";
 };
 
 &spi0 {
-	status = "disabled";
-	assigned-clocks = <&cru CLK_SPI0>;
-	assigned-clock-rates = <200000000>;
-	num-cs = <2>;
+	// status = "disabled";
+    status = "okay";
+    assigned-clocks = <&cru CLK_SPI0>;
+    assigned-clock-rates = <200000000>;
+    num-cs = <2>;
+    pinctrl-names = "default";
+    pinctrl-0 = <&spi0m2_pins &spi0m2_cs0 &spi0m2_cs1>;
 };
 
 &spi1 {
@@ -1084,55 +856,46 @@ &spi1 {
 	assigned-clock-rates = <200000000>;
 };
 
-&uart4 {
+&pwm0 {
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm0m2_pins>;
+	status = "okay";
+};
+
+&pwm1 {
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm1m2_pins>;
 	status = "disabled";
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart4m2_xfer>;
 };
 
-&uart9 {
+&pwm3 {
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm3m3_pins>;
 	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart9m2_xfer &uart9m2_ctsn>;
 };
 
-&i2c1 {
-	status = "okay";
+&pwm11 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&i2c1m2_xfer>;
-
-	aw87xxx_pa1: aw87xxx_pa1@59 {
-		compatible = "awinic,aw87xxx_pa";
-		#sound-dai-cells = <0>;
-		reg = <0x59>;
-		dev_index = <0>;
-		status = "okay";
-	};
-
-	aw87xxx_pa2: aw87xxx_pa2@5a {
-		compatible = "awinic,aw87xxx_pa";
-		#sound-dai-cells = <0>;
-		reg = <0x5a>;
-		dev_index = <1>;
-		status = "okay";
-	};
+	pinctrl-0 = <&pwm11m2_pins>;
+	status = "disabled";
 };
 
-&i2c4 {
+&pwm13 {
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm13m2_pins>;
 	status = "disabled";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c4m3_xfer>;
 };
 
-&i2c5 {
-	status = "disabled";
-	pinctrl-names = "default";
-	pinctrl-0 = <&i2c5m3_xfer>;
+&pwm14 {
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm14m0_pins>;
+	status = "okay";
 };
 
-&pwm3 {
-	status = "disabled";
-	pinctrl-0 = <&pwm3m3_pins>;
+&pwm15 {
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm15m2_pins>;
+	status = "okay";
 };
 
 &uart1 {
@@ -1140,8 +903,18 @@ &uart1 {
 	pinctrl-0 = <&uart1m1_xfer>;
 };
 
-&uart6 {
+&uart3 {
+	status = "okay";
+	pinctrl-0 = <&uart3m0_xfer>;
+};
+
+&uart4 {
 	status = "disabled";
+	pinctrl-0 = <&uart4m2_xfer>;
+};
+
+&uart6 {
+	status = "okay";
 	pinctrl-0 = <&uart6m2_xfer>;
 };
 
@@ -1150,6 +923,28 @@ &uart7 {
 	pinctrl-0 = <&uart7m2_xfer>;
 };
 
+&uart9 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart9m2_xfer &uart9m2_ctsn>;
+};
+
+&can1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&can1m1_pins>;
+	assigned-clocks = <&cru CLK_CAN1>;
+	assigned-clock-rates = <200000000>;
+};
+
+&can2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&can2m0_pins>;
+	assigned-clocks = <&cru CLK_CAN2>;
+	assigned-clock-rates = <200000000>;
+};
+
 /* delete gpio0_c2 */
 &pmic_pins {
 	rockchip,pins =
@@ -1167,45 +962,77 @@ &pmic_pins {
 		<0 RK_PD6 1 &pcfg_pull_none>;
 };
 
-&pwm11 {
-	compatible = "rockchip,remotectl-pwm";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pwm11m2_pins>;
-	remote_pwm_id = <3>;
-	handle_cpu_id = <1>;
-	remote_support_psci = <0>;
-	status = "okay";
-
-	ir_key1 {
-		rockchip,usercode = <0xfb04>;
-		rockchip,key_table =
-			<0xa3   KEY_ENTER>,
-			<0xe4   388>,
-			<0xf5   KEY_BACK>,
-			<0xbb   KEY_UP>,
-			<0xe2   KEY_DOWN>,
-			<0xe3   KEY_LEFT>,
-			<0xb7   KEY_RIGHT>,
-			<0xe0   KEY_HOME>,
-			<0xba   KEY_VOLUMEUP>,
-			<0xda   KEY_VOLUMEUP>,
-			<0xe6   KEY_VOLUMEDOWN>,
-			<0xdb   KEY_VOLUMEDOWN>,
-			<0xbc   KEY_SEARCH>,
-			<0xb2   KEY_POWER>,
-			<0xe5   KEY_POWER>,
-			<0xde   KEY_POWER>,
-			<0xdc   KEY_MUTE>,
-			<0xa2   KEY_MENU>,
-			<0xec   KEY_1>,
-			<0xef   KEY_2>,
-			<0xee   KEY_3>,
-			<0xf0   KEY_4>,
-			<0xf3   KEY_5>,
-			<0xf2   KEY_6>,
-			<0xf4   KEY_7>,
-			<0xf7   KEY_8>,
-			<0xf6   KEY_9>,
-			<0xb8   KEY_0>;
+&pinctrl {
+	usb-typec {
+		usbc0_int: usbc0-int {
+			rockchip,pins = <4 RK_PA3 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+
+		typec5v_pwren: typec5v-pwren {
+			rockchip,pins = <4 RK_PA4 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
 	};
+
+	usb {
+		vcc5v0_host_en: vcc5v0-host-en {
+			rockchip,pins = <4 RK_PA4 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	sdio-pwrseq {
+		wifi_enable_h: wifi-enable-h {
+			rockchip,pins = <0 RK_PD0 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	hym8563 {
+		hym8563_int: hym8563-int {
+			rockchip,pins = <0 RK_PB0 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+
+	wireless-bluetooth {
+		uart9_gpios: uart9-gpios {
+			rockchip,pins = <3 RK_PD2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		bt_reset_gpio: bt-reset-gpio {
+			rockchip,pins = <0 RK_PD3 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		bt_wake_gpio: bt-wake-gpio {
+			rockchip,pins = <0 RK_PC6 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		bt_irq_gpio: bt-irq-gpio {
+			rockchip,pins = <0 RK_PC5 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
+	};
+
+	wireless-wlan {
+		wifi_host_wake_irq: wifi-host-wake-irq {
+			rockchip,pins = <0 RK_PA0 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
+	};
+
+	sata {
+		sata_reset:sata-reset{
+			rockchip,pins = <3 RK_PD1 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+
+	gpio-fan {
+        fan0_en: fan0-en {
+            rockchip,pins = <1 RK_PB7 RK_FUNC_GPIO &pcfg_pull_none>;
+        };
+        fan1_en: fan1-en {
+            rockchip,pins = <1 RK_PB6 RK_FUNC_GPIO &pcfg_pull_none>;
+        };
+        fan2_en: fan2-en {
+            rockchip,pins = <1 RK_PA3 RK_FUNC_GPIO &pcfg_pull_none>;
+        };
+        fan3_en: fan3-en {
+            rockchip,pins = <1 RK_PD2 RK_FUNC_GPIO &pcfg_pull_none>;
+        };
+    };
 };
