--- a/drivers/rtc/rtc-hym8563.c
+++ b/drivers/rtc/rtc-hym8563.c
@@ -14,6 +14,7 @@
 #include <linux/i2c.h>
 #include <linux/bcd.h>
 #include <linux/rtc.h>
+#include <linux/delay.h>
 
 #define HYM8563_CTL1		0x00
 #define HYM8563_CTL1_TEST	BIT(7)
@@ -467,39 +468,54 @@
 static int hym8563_init_device(struct i2c_client *client)
 {
 	int ret;
+	int retry_count = 10;
 
-	ret = i2c_smbus_read_byte_data(client, HYM8563_CTL1);
-	if (ret < 0)
-		dev_err(&client->dev, "%s: error read i2c data %d\n",
-			__func__, ret);
-
-	/* Clear stop flag if present */
-	ret = i2c_smbus_write_byte_data(client, HYM8563_CTL1, 0);
-	if (ret < 0)
-		return ret;
-
-	ret = i2c_smbus_read_byte_data(client, HYM8563_CTL2);
-	if (ret < 0)
-		return ret;
-
-	/* Disable alarm and timer interrupts */
-	ret &= ~HYM8563_CTL2_AIE;
-	ret &= ~HYM8563_CTL2_TIE;
-
-	/* Clear any pending alarm and timer flags */
-	if (ret & HYM8563_CTL2_AF)
-		ret &= ~HYM8563_CTL2_AF;
-
-	if (ret & HYM8563_CTL2_TF)
-		ret &= ~HYM8563_CTL2_TF;
-
-	ret &= ~HYM8563_CTL2_TI_TP;
+	while (retry_count--) {
+		ret = i2c_smbus_read_byte_data(client, HYM8563_CTL1);
+		if (ret < 0)
+			dev_err(&client->dev, "%s: error read i2c data %d\n",
+				__func__, ret);
+
+		/* Clear stop flag if present */
+		ret = i2c_smbus_write_byte_data(client, HYM8563_CTL1, 0);
+		if (ret < 0)
+			goto retry;
+
+		ret = i2c_smbus_read_byte_data(client, HYM8563_CTL2);
+		if (ret < 0)
+			goto retry;
+
+		/* Disable alarm and timer interrupts */
+		ret &= ~HYM8563_CTL2_AIE;
+		ret &= ~HYM8563_CTL2_TIE;
+
+		/* Clear any pending alarm and timer flags */
+		if (ret & HYM8563_CTL2_AF)
+			ret &= ~HYM8563_CTL2_AF;
+
+		if (ret & HYM8563_CTL2_TF)
+			ret &= ~HYM8563_CTL2_TF;
+
+		ret &= ~HYM8563_CTL2_TI_TP;
+
+		/* Reset timer cnt and Set timer countdown 1s per count */
+		i2c_smbus_write_byte_data(client, HYM8563_TMR_CNT, 0);
+		i2c_smbus_write_byte_data(client, HYM8563_TMR_CTL, HYM8563_TMR_CFG);
+
+		ret = i2c_smbus_write_byte_data(client, HYM8563_CTL2, ret);
+		if (ret < 0)
+			goto retry;
 
-	/* Reset timer cnt and Set timer countdown 1s per count */
-	i2c_smbus_write_byte_data(client, HYM8563_TMR_CNT, 0);
-	i2c_smbus_write_byte_data(client, HYM8563_TMR_CTL, HYM8563_TMR_CFG);
+		return 0;
 
-	return i2c_smbus_write_byte_data(client, HYM8563_CTL2, ret);
+retry:
+		if (retry_count) {
+			dev_err(&client->dev, "%s: error retry=%d, ret=%d\n",
+				__func__, 10 - retry_count, ret);
+			msleep(20);
+		}
+	}
+	return ret;
 }
 
 #ifdef CONFIG_PM_SLEEP
